src/Github/Api/V3/CreateReleaseThroughApiCall.php

```diff
@@ -34,32 +40,40 @@ public function __invoke(
         $request = $this->messageFactory
             ->createRequest(
                 'POST',
-                self::API_ROOT . 'repos/' . $repository->owner() . '/' . $repository->name() . '/releases',
+                sprintf(self::API_URI, $repository->owner(), $repository->name()),
             )
             ->withAddedHeader('Content-Type', 'application/json')
             ->withAddedHeader('User-Agent', 'Ocramius\'s minimal API V3 client')
             ->withAddedHeader('Authorization', 'token ' . $this->apiToken);
 
-        $request
-            ->getBody()
-            ->write(Json\encode([
-                'tag_name' => $version->fullReleaseName(),
-                'name'     => $version->fullReleaseName(),
-                'body'     => $releaseNotes,
-            ]));
+        $request->getBody()->write(encode([
+            'tag_name' => $version->fullReleaseName(),
+            'name'     => $version->fullReleaseName(),
+            'body'     => $releaseNotes,
+        ]));
 
         $response = $this->client->sendRequest($request);
```

Comment:
BTW, IIRC we already do provide an HTTP client with logging? 🤔

Reply:
This patch extracts semi-meaningful (  [`message` field describing the error](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#client-errors)  ) from the HTTP Response we receive back from GitHub API.

Currently, we're **throwing on failed API status codes** and **only logging applications-level events** (eg. No change logs found).

Reply:
Fair enough - good improvement to be added 👍

Reply:
> only logging applications-level events

BTW, we log full request/response when debug level is raised